NEW: device settings to synchronize the spa clock with Homey

oh2th opened the pull request:
- new balboa api setTime()
- new device interval onClockSyncInterval
- new device method setClock to call the api
- configured in device settings
- no flow cards, no capabilites

Dutch is auto-translated - need verification in settings.json

oh2th committed:
Add setTime(date, time, military_time) for api
And use it in the device for setClock() to periodically set the time on the spa.
Will keep the spa in time after a poeer outage and with DST.
Configurable from device settings for
- enable the sync
- 24h/12h time setting (military_time)
- sync interval hours 1-24.

oh2th committed:
setTime ready after testing
- new api setTime()
- new device interval onClockSyncInterval
- new device method setClock call the api
- configured in device settings, dutch is auto-translated - need verification in settings.json
- no flow cards, no capabilites

oh2th committed:
README update for setTime

martijnpoppen commented on drivers/main-device.js:
```diff
@@ -351,16 +351,33 @@ module.exports = class mainDevice extends Homey.Device {
             if (this.getAvailable()) {
                 await this.setCapabilityValuesInterval(settings.update_interval);
+                // Clock sync is disable if clock_sync_interval = 0
+                if(settings.clock_sync_interval !== 0) await this.setClockSyncInterval(settings.clock_sync_interval);
```
I'm a bit against setting a extra interval, this will increase the RAM usage of the app by quite a lot.
I would rather check if the time inside the current interval is close to the selected time.

Something like:

```
current time = 24

if(current_time % interval === 0) {
   setTime()
}
```

oh2th replied:
This would make it simpler. Will check an refactor it.

oh2th replied:
Complete refactoring and integrate the setTime function to setCapabilityValues(). Now setTime is trigger when needed.
- If Spa time is not set or is off by 5 minutes
- Device settings on now on/off for enabling clock sync. Default is off, so doesn't change existing devices.

martijnpoppen commented on .homeycompose/drivers/templates/settings.json:
```diff
@@ -59,6 +59,47 @@
+          },
+          "hint":{
+            "en": "If enabled, the spa clock will be set to 24h format, otherwise 12h format.",
+            "nl": "Indien ingeschakeld, zal de spa klok ingesteld worden op 24-uurs formaat, anders 12-uurs formaat."
```
👍🏼

oh2th committed:
Removed setClockInterval and refactored clock logic
- setClock logic now within normal onPollInterval
- found that Spa reports its time as hour and minute
- compare if differs more than 5 minutes
- take action to setTime only when needed and allowed by configuration.

settings.json updated for clock_sync boolean.
Please verify dutch translation.

vscode reformated some areas where spaces were missing.

oh2th committed:
Sync clock also when militaty time is changed
- clock_24h != military

oh2th commented:
Screenshot of the settings in device's Advanced Settings.
![IMG_CD7ACAE60A72-1](https://user-images.githubusercontent.com/59442255/215462266-80adb4b3-6e9b-41bd-8c3c-bed64ade70e1.jpeg)

oh2th commented:
One caveat, if a users changes the 24h clock from the CMS app, this app will return it back to what was expected by device settings.

oh2th committed:
One more think, spa should be online to setTime.
Optimised the if conditions for setTime
